src/blog/2024/03/looking-towards-node-red-4.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Looking towards Node-RED 4.0 and beyond
+subtitle: A look at what is coming in Node-RED 4.0
+description: A look at what is coming in Node-RED 4.0
+date: 2024-03-06
+authors: ["nick-oleary"]
+image: /blog/2024/03/images/looking-towards-node-red-4.png
+tags:
+    - posts
+    - flowfuse
+    - node-red
+---
+
+With Node-RED 4.0 coming soon, I wanted to take a look at what users can expect to see with the new release,
+as well as some of the new features we're working on.
+
+<!--more-->
+
+The Node-RED project [schedules its releases](https://nodered.org/about/releases/) around a yearly major release that coincides with when a Node.js version
+reaches its end-of-support. This lets us drop support for that node.js version and update the default version of node used
+in the docker containers we publish. We treat this as a major change because  it might require actions on the end-users part to update any additional modules they have installed.
+
+With Node-RED 4.0, we will be dropping support for anything earlier than Node 18 - with Node 20 the currently recommended version to use. That will give users almost 2 full years before needing to consider another Node.js upgrade.
+
+Now, talking about Node.js versions is not that exciting. What is more exciting is to look at what new features are coming to Node-RED.
+
+There are a few things already merged and ready to be released in the first beta release, and more landing each week.
+
+### More auto-complete
+
+Node-RED already has simple auto-complete on `msg` fields in the editor. We've now extended that to also work with `flow`/`global` context inputs as well as the `env` type for accessing environment variables.
+
+![](./images/nr4-auto-complete.png)
+
+This makes is so much easier to work with these types of properties - being sure you're using something that exists rather than having to switch between different views in the editor to get the names right.
```

Comment:
```suggestion
This makes it so much easier to work with these types of properties - being sure you're using something that exists rather than having to switch between different views in the editor to get the names right.
```